server/src/tram_track_graph_transformer/tram_track_graph_transformer.py

```diff
@@ -191,6 +192,16 @@ def _get_new_node_id(self):
         self._max_node_id += 1
         return self._max_node_id
 
+    def _add_edge(self, graph: nx.DiGraph, first_node: Node, second_node: Node):
```

Comment:
This method doesn't use `self` anywhere, so it could be a class method.

I believe the method name could be more descriptive, as inside of it we're doing something more than just adding an edge, something like `_add_geodetic_edge` would be nice.

Also as we're dealing with a directed graph, I think a better name for these arguments would be something like `source_node` and `dest_node`, which would make it clear what will be the direction of the added edge. 

```suggestion
    def _add_geodetic_edge(self, graph: nx.DiGraph, source_node: Node, dest_node: Node):
```

Reply:
`_interpolate_path_nodes` also doesn't use self. Maybe let it be class method too?